### ISSUE TITLE
Save highlight location/position in elastic for sorting

### ISSUE THREAD

**Issue by sywhb** [omnivore-app/omnivore#1056]

**Comment by sywhb** [comment]
This is to be done in the frontend or backend

**Comment by jacksonh** [comment]
@sywhb I think frontend can pretty easily pass a value to the backend in the Create highlight API if we add a new optional position field. Maybe it should just be a 0-100 percentage.

**Comment by sywhb** [comment]
@jacksonh Yes, I will add that

**Comment by jacksonh** [comment]
@sywhb can we make it take two values, one a number percentage, the other an anchorIndex. So should look just like the items to save reading progresss:

```
    highlightPositionPercent: Float
    highlightPositionAnchorIndex: Int
```

both should be optional

**Comment by sywhb** [comment]
@jacksonh yes, is it enough for articles with multiple columns?

**Comment by sywhb** [comment]
Likely PDF files

**Comment by sywhb** [comment]
For example, two highlights on the same horizontal level but in different vertical columns

**Comment by jacksonh** [comment]
Yeah, i think in that case it wont be very good, but mostly its useful to give users an idea of where the quote is in a document. Until we try doing PDF readability i guess we can't do much better.

**Comment by sywhb** [comment]
<img width="1445" alt="Screenshot 2022-10-18 at 12 31 54 PM" src="https://user-images.githubusercontent.com/12963965/196336019-af9a0de3-689a-48bc-8d20-6fa243f4d5f8.png">

**Comment by jacksonh** [comment]
I will look closer at the API and see if there is more detail we get, but I think its just treated as three rectangles in this case.

**Comment by sywhb** [comment]
@jacksonh I have looked at the API previously and did a hack in the plugin

The API gives four value indicating the box position
```export const getHighlightPoint = (patch: string): HighlightPoint => {
  const { bbox } = JSON.parse(patch) as { bbox: number[] }
  if (!bbox || bbox.length !== 4) {
    return { left: 0, top: 0 }
  }
  return { left: bbox[0], top: bbox[1] }
}
```

So I just compare them vertically and then horizontally but this is just a hack
```
export const compareHighlightsInFile = (a: Highlight, b: Highlight): number => {
  // get the position of the highlight in the file
  const highlightPointA = getHighlightPoint(a.patch)
  const highlightPointB = getHighlightPoint(b.patch)
  if (highlightPointA.top === highlightPointB.top) {
    // if top is same, sort by left
    return highlightPointA.left - highlightPointB.left
  }
  // sort by top
  return highlightPointA.top - highlightPointB.top
}```

**Comment by jacksonh** [comment]
Yeah maybe we could use that same hack to calculate a percentage on the client.

**Comment by sywhb** [comment]
We could use a library to sort the highlights by positional value

**Comment by sywhb** [comment]
API update: #1346

**Comment by jacksonh** [comment]
Think we are all good here now/
